shell/browser/api/electron_api_system_preferences_win.cc

```diff
@@ -117,6 +166,24 @@ std::string SystemPreferences::GetColor(gin_helper::ErrorThrower thrower,
   return ToRGBHex(color_utils::GetSysSkColor(id));
 }
 
+std::string SystemPreferences::GetMediaAccessStatus(
+    const std::string& media_type,
+    gin_helper::Arguments* args) {
+  if (media_type == "camera") {
+    return ConvertDeviceAccessStatus(
+        GetDeviceAccessStatus(DeviceClass::DeviceClass_VideoCapture));
+  } else if (media_type == "microphone") {
+    return ConvertDeviceAccessStatus(
+        GetDeviceAccessStatus(DeviceClass::DeviceClass_AudioCapture));
+  } else if (media_type == "screen") {
+    return ConvertDeviceAccessStatus(
+        DeviceAccessStatus::DeviceAccessStatus_Allowed);
+  } else {
+    args->ThrowError("Invalid media type");
```

Comment:
Not originating from this PR but when we have a `unknown` state in the docs, why do we want to throw an error ? seem to do it for macOS as well.

Reply:
@deepak1556 The return type can have `unknown`, the `media_type` argument can't be `unknown`

Reply:
For an unknown media type why can't the return type by unknown, the current system calls both on windows and macOS never return an unknown state, the value is being unused. I wouldn't mind if this state wasn't documented but it is :)

Reply:
Would be good to bring this up with api-WG

Reply:
I can address it in a separate PR.

Reply:
Thanks!